src/js/apiData.js

```diff
@@ -1,5 +1,136 @@
+  resultArray.push(meteo);
+  const dividedArray = divideArray(jsonForecast.list);
+  for(let i = 1; i < dividedArray.length; i++){
+      resultArray.push(transformToObject(dividedArray[i]));
+  }
+  return resultArray;
+}
+
+
+function divideArray(meteoArr){
+  var dayArray = [];
+  var dividedArray = [];
+
+  let data1 = new Date(meteoArr[0].dt_txt);
+  data1 = data1.getDay();
+  dayArray.push(meteoArr[0])
+
+  for(let i = 1; i < meteoArr.length; i++){
+      let data2 = new Date(meteoArr[i].dt_txt);
+      data2 = data2.getDay();
+      if(data1 == data2){
+          dayArray.push(meteoArr[i])
+      }
+      else {
+          dividedArray.push(dayArray);
+          dayArray = [];
+          dayArray.push(meteoArr[i]);
+      }
+      data1 = data2;
+
+  }
+  dividedArray.push(dayArray);
+  return dividedArray;
+}
+
+function transformToObject(dayArray){
+  let date = dayArray[0].dt_txt;
+  let temp_min = dayArray[0].main.temp_min;
+  let temp_max = dayArray[0].main.temp_max;
+  let weatherArray = [dayArray[0].weather[0].main];
+  let descriptionArray = [dayArray[0].weather[0].description];
+  let iconArray = [dayArray[0].weather[0].icon];
+  let sumTemp = 0;
+  let sumHumidity = 0;
+  let sumPressure = 0;
+  let sumWindSpeed = 0;
+  for(let i = 1; i < dayArray.length; i++){
+      if(dayArray[i].main.temp_min < temp_min)
+          temp_min = dayArray[i].main.temp_min;
+      if(dayArray[i].main.temp_max > temp_max)
+          temp_max = dayArray[i].main.temp_max;
+      weatherArray.push(dayArray[i].weather[0].main);
+      descriptionArray.push(dayArray[i].weather[0].description);
+      iconArray.push(dayArray[i].weather[0].icon);
+      sumTemp += dayArray[i].main.temp;
+      sumHumidity += dayArray[i].main.humidity;
+      sumPressure += dayArray[i].main.pressure;
+      sumWindSpeed += dayArray[i].wind.speed;
+  }
+  let weather = mostCommonInArray(weatherArray);
+  let description = mostCommonInArray(descriptionArray);
+  let icon = mostCommonInArray(iconArray);
+  let temp = Math.floor(sumTemp / dayArray.length);
+  let humidity = Math.floor(sumHumidity / dayArray.length);
+  let pressure = Math.floor(sumPressure / dayArray.length);
+  let windSpeed = Math.floor(sumWindSpeed / dayArray.length);
+
+  return toMeteoObject(date,temp,Math.floor(temp_min),Math.floor(temp_max),weather,description,icon,windSpeed,pressure,humidity);
+}
+
+
+function toMeteoObject(date,temp,temp_min,temp_max,weather, description,icon,wind_speed, pressure, humidity){
+  const object = {
+      weather: weather,
+      description: description,
+      icon: icon,
+      temp_min: temp_min,
+      temp_max: temp_max,
+      wind_speed: wind_speed,
+      pressure: pressure,
+      humidity: humidity,
+      date: date,
+      temp: temp,
+  }
+  return object;
+}
```

Comment:
to nie wyglada dobrze - jesli juz chcesz cos takiego zrobic to lepiej stworzyc klase do tego, ale nadal pisanie funkcji z milionem argumentow brzmi jak zly pomysl